src/PagesBody/DogDataPage/DogDataPage.js

```diff
@@ -1,8 +1,52 @@
 import DogData from '../../Organisms/DogData/DogData';
-const randomDogID = 'e8085155-6b82-41f4-a5e5-b4745c332ce2';
+import { useContext, useEffect, useState } from 'react';
+import { useLocation, useParams } from 'react-router-dom';
+import { ParticipantContext } from '../../Context/ParticipantContext';
+
+// @TODO delete this in future
+const fakeDogId = 'VIII-40407';
 
 const DogDataPage = () => {
-  return <DogData id={randomDogID}></DogData>;
+  const [isPending, setIsPending] = useState(true);
+
+  // From the previous page 'dog-submit' after choosing dog we passed to this dog-data page
+  // dogId by state passed by navigate
+  //      navigate(`./dog-data`, {
+  //         state: { dogId: '' },
+  //       });
+
+  const location = useLocation();
+  const [dogId, setDogId] = useState(null);
+  const { dogId: dogIdParams } = useParams();
+  const { participantData } = useContext(ParticipantContext);
+
+  /*
+  These works three-way:
+    1) It checks if somebody pass dogId by navigate from previous page
+    2) It checks if somebody pass dogId by url address (useParams())
+    3) It checks participantContext if participant have only 1 dog (so pass this dogId)
+    4) I pass fake dogId.
+   */
```

Comment:
To consider - maybe we should change path? from "/dog-data" to "/dog-data/:dogId"?
First version of paths was created long time ago, and if there is a need we always can change them. For example, we changed /user to /user/:userId and /:contest-details to constests/:contestId

Reply:
so let's do that :)